net/peer_tracker_test.go

```diff
@@ -0,0 +1,85 @@
+package net_test
+
+import (
+	"context"
+	"sort"
+	"testing"
+
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p/p2p/net/mock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/go-filecoin/net"
+	th "github.com/filecoin-project/go-filecoin/testhelpers"
+	tf "github.com/filecoin-project/go-filecoin/testhelpers/testflags"
+	"github.com/filecoin-project/go-filecoin/types"
+)
+
+func TestPeerTrackerTracks(t *testing.T) {
+	tf.UnitTest(t)
+
+	tracker := net.NewPeerTracker()
+	pid0 := th.RequireIntPeerID(t, 0)
+	pid1 := th.RequireIntPeerID(t, 1)
+	pid3 := th.RequireIntPeerID(t, 3)
+	pid7 := th.RequireIntPeerID(t, 7)
+
+	ci0 := types.NewChainInfo(pid0, types.NewTipSetKey(), 0)
+	ci1 := types.NewChainInfo(pid1, types.NewTipSetKey(), 0)
+	ci3 := types.NewChainInfo(pid3, types.NewTipSetKey(), 0)
+	ci7 := types.NewChainInfo(pid7, types.NewTipSetKey(), 0)
+
+	tracker.Track(ci0)
+	tracker.Track(ci1)
+	tracker.Track(ci3)
+	tracker.Track(ci7)
+
+	tracked := tracker.List()
+	sort.Sort(types.CISlice(tracked))
+	expected := []*types.ChainInfo{ci0, ci1, ci3, ci7}
+	sort.Sort(types.CISlice(expected))
+	assert.Equal(t, expected, tracked)
+}
+
+func TestPeerTrackerDisconnect(t *testing.T) {
+	tf.UnitTest(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
```

Comment:
not sure this is needed, but I haven't used mocknet before.

Reply:
It's not strictly needed to test the tracker API but it would be annoying to get connections without it.

It's nice to have here because it ensures that the callback signature fits into the libp2p callback system (and that the callingback works)